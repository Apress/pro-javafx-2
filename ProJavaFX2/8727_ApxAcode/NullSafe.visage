/*
 * @test
 * @run
 */

class A {
  var objVar:Object;
  var intVar:Integer;
  var stringVar:String;

  function retObj():Object {
    println("should not get called");
    objVar
  }
}

var nullA:A = null;

println("nullA.objVar = {nullA.objVar}");
println("nullA.intVar = {nullA.intVar}");
println("nullA.stringVar = {nullA.stringVar}");
println("nullA.retObj() = {nullA.retObj()}");
