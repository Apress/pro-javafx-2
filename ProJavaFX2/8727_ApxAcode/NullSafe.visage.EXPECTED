nullA.objVar = null
nullA.intVar = 0
nullA.stringVar = 
nullA.retObj() = null
